src/TicketServiceProvider.php

```diff
@@ -3,22 +3,28 @@
 namespace dnj\Ticket;
 
 use Illuminate\Support\ServiceProvider;
+use dnj\Filesystem\Contracts\IFile;
+use dnj\Filesystem\Local\File;
 
 class TicketServiceProvider extends ServiceProvider
 {
     public function register()
     {
-        $this->mergeConfigFrom(__DIR__.'/../config/ticket.php', 'ticket');
+        $this->mergeConfigFrom(__DIR__ . '/../config/ticket.php', 'ticket');
     }
 
     public function boot()
     {
-        $this->loadRoutesFrom(__DIR__.'/../routes/api.php');
-        $this->loadMigrationsFrom(__DIR__.'/../database/migrations');
+        $this->loadRoutesFrom(__DIR__ . '/../routes/api.php');
+        $this->loadMigrationsFrom(__DIR__ . '/../database/migrations');
         if ($this->app->runningInConsole()) {
             $this->publishes([
-                __DIR__.'/../config/ticket.php' => config_path('ticket.php'),
+                __DIR__ . '/../config/ticket.php' => config_path('ticket.php'),
             ], 'config');
         }
+
+        $this->app->bind(IFile::class, function ($app) {
+            return new File(base_path() . '/' . config('ticket.bucket') . '/file');
```

Comment:
I really got confused understanding your goal here!
Why you did this? A `IFile` is a representor a physical file and we use interfaces instead of concrete classes because we don't know of underneath storage layer.
I suggest to put a attachment root in config file as `IDirectory` and use it's `file()` method to access it's files.